### ISSUE TITLE
[Bug]: Transformations behaviour possibly not expected when predicting with fresh dataloader

### ISSUE THREAD

**Issue by blaz-r** [open-edge-platform/anomalib#2254]
### Describe the bug

Inside the engine code the trasnforms from the datamodule and dataloader are taken before the ones from the model:
https://github.com/openvinotoolkit/anomalib/blob/2bd2842ec33c6eedb351d53cf1a1082069ff69dc/src/anomalib/engine/engine.py#L382-L398

This can cause problems if datamodule was never used inside the trainer. In that case looking at the following code, the transforms returned are just resize (since there is no trainer to take the correct model transforms  from). This leads to missing the potential normalization from the model:
https://github.com/openvinotoolkit/anomalib/blob/2bd2842ec33c6eedb351d53cf1a1082069ff69dc/src/anomalib/data/base/datamodule.py#L266-L277

This happens only if you are calling setup and dataloader from the datamodule outside trainer (as that is not set in this case) like this:
```python
datamodule= MVTec(..)
datamodule.setup() # this calls self.train_transforms mentioned above
datamodule.test_dataloader() # only resize transofmr here
```
There is also code below to reproduce this.

A workaround I have at the moment is doing this:
```python
dataloader.dataset.transform = None
```

This way the dataloader transforms are ignored and the ones from model are taken. Another possibility that I see is by manually setting the transforms of the datamodule with `model.configure_transforms(image_size)`.

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

```python
    #### CONFIGURE THIS ####
    mvtec_path = "../datasets/MVTec"
    #####

    data = MVTec(root=mvtec_path, image_size=(42, 42), category="bottle", num_workers=0)
    data.setup()

    print(data.test_dataloader().dataset.transform)
```

### OS information

OS information:
- OS: Win11
- Python version: 3.10.3
- Anomalib version: 1.2.0dev
- PyTorch version: 2.3.0
- CUDA/cuDNN version: x
- GPU models and configuration: x
- Any other relevant information: x

### Expected behavior

I would expect that in this case, the model transforms would take priority over the ones in dataloader, but I see how this would cause trouble in case of custom transforms inside datamodule.

### Pip/GitHub

GitHub

### What version/branch did you use?

1.2.0dev

### Configuration YAML

```yaml
/
```

### Logs

```shell
/
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
Is there any updates on this one?

**Comment by samet-akcay** [comment]
@blaz-r, not to address this one specifically, but @djdameln is working on some changes, which might address this

**Comment by blaz-r** [comment]
Okay. I'll use the workaround for now and we'll see when those changes are added.

**Comment by blaz-r** [comment]
This is no longer problematic with v2 (I think), so I'm closing the issue.

**Comment by samet-akcay** [comment]
Actually, there are quite a lot of issues to be closed as v2 addresses majority of them. We just need to update them
